haystack/components/joiners/branch.py

```diff
@@ -0,0 +1,147 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+from typing import Any, Dict
+
+from haystack import component, default_from_dict, default_to_dict, logging
+from haystack.core.component.types import Variadic
+from haystack.utils import deserialize_type, serialize_type
+
+if sys.version_info < (3, 10):
+    from typing_extensions import TypeAlias
+else:
+    from typing import TypeAlias
+
+
+logger = logging.getLogger(__name__)
+
+
+@component(is_greedy=True)
+class BranchJoiner:
+    """
+    A component to join different branches of a pipeline into one single output.
+
+    `BranchJoiner` receives multiple data connections of the same type from other components, and passes the first
+    value coming to its single output, possibly distributing it to various other components.
```

Comment:
I'm not sure about the distributing part here, didn't we agree to avoid mentioning this avoid confusing people?

Reply:
yeah I recycled some text, let me fix